cmd/oceantv/broadcast_manager.go

```diff
@@ -177,11 +172,26 @@ func (m *OceanBroadcastManager) StopBroadcast(ctx Ctx, cfg *Cfg, store Store, sv
 	return stopBroadcast(ctx, cfg, store, svc, m.log)
 }
 
-// SaveBroadcast saves a broadcast to the datastore.
-// It uses AusOcean methods for saving, and updating the vidforward service if
-// configuration if in use.
-func (m *OceanBroadcastManager) SaveBroadcast(ctx Ctx, cfg *Cfg, store Store) error {
-	return saveBroadcast(ctx, cfg, store, m.log)
+// Save performs broadcast configuration update operations.
+// If ctx is nil, the background context will be used.
+//
+// update allows for the update of specific fields. After this update takes
+// place, the config we currently point at will be updated with any changes
+// that we have applied, and with anything from the store before the update.
+// If this is nil, the config currently in store will be replaced.
+func (m *OceanBroadcastManager) Save(ctx Ctx, update func(_cfg *Cfg)) error {
+	_ctx := context.Background()
```

Comment:
Why set _ctx to Background, only to overwrite it if ctx is nil? Why not just the following?

```
	if ctx != nil {
		ctx = context.Background()
	}
```

Reply:
It gets overwritten if ctx is _not_ nil i.e. if the caller has provided a context then we use that, otherwise we just use context.Background().

Reply:
That's exactly what my code snippet attempt to do, i.e., use ctx unless it is nil, in which case use the background context.

Reply:
Actually, I meant this:

```
if ctx == nil {
		ctx = context.Background()
	}
```

Reply:
Oh yeah, you're right